### DISCUSSION TITLE
/adding-guests-to-an-office-365-group-via-the-microsoft-graph-api/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#244] [Comments]
Adding guests to an Office 365 Group via the Microsoft Graph API
        
[https://www.eliostruyf.com/adding-guests-to-an-office-365-group-via-the-microsoft-graph-api/](https://www.eliostruyf.com/adding-guests-to-an-office-365-group-via-the-microsoft-graph-api/)

**Comment by comment-migrator** [comment]
```header
Originally posted by slava on 2017-01-20 21:57
```

Great! Thanks!

\--Slava

**Comment by comment-migrator** [comment]
```header
Originally posted by Simon on 2017-04-27 18:32
```

Hi Elio, thanks for the sample. I was able to reproduce everything and the invitation ships but the customized body does not seem to take.  
{  
"invitedUserDisplayName": "User Four",  
"invitedUserEmailAddress": "user4@domain.onmicrosft.com",  
"inviteRedirectUrl": "[https://microsoft.com/](https://microsoft.com/ "https://microsoft.com/")",  
"sendInvitationMessage": true,  
"invitedUserMessageInfo": {"customizedMessageBody": "Hi Elio, you can find more information here: [https://www.eliostruyf.com](https://www.eliostruyf.com "https://www.eliostruyf.com")"}  
}  
Everything in here works except invitedUserMessageInfo and customizedMessageBody. Have you had to change this yourself? I cannot find proper documentation of the api that lists those potential parameters.

Thanks,

Simon

**Comment by comment-migrator** [comment]
```header
Originally posted by Simon on 2017-04-27 20:08
```

Hi Elio,  
thanks for the sample, it helped us getting the invitation process automated.  
One thing we haven't been able to make work is the invitation message customization.  
{  
"invitedUserDisplayName": "User Four",  
"invitedUserEmailAddress": "user4@domain.onmicrosft.com",  
"inviteRedirectUrl": "[https://microsoft.com/](https://microsoft.com/ "https://microsoft.com/")",  
"sendInvitationMessage": true,  
"invitedUserMessageInfo": {"customizedMessageBody": "Hi Elio, you can find more information here: [https://www.eliostruyf.com](https://www.eliostruyf.com "https://www.eliostruyf.com")"}  
}

With this invitation for example everything works well except for invitedUserMessageInfo and customizedMessageBody. Have you seen this problem?

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-05-03 11:55
```

Not yet experienced any problems with it. API is in beta so it could be that they are changing things internally. Definitely check:  
\- [https://developer.microsoft...](https://developer.microsoft.com/en-us/graph/docs/api-reference/beta/resources/invitedusermessageinfo "https://developer.microsoft.com/en-us/graph/docs/api-reference/beta/resources/invitedusermessageinfo")  
\- [https://developer.microsoft...](https://developer.microsoft.com/en-us/graph/docs/api-reference/beta/resources/invitation "https://developer.microsoft.com/en-us/graph/docs/api-reference/beta/resources/invitation")

**Reply by comment-migrator** [reply]
```header
Originally posted by Simon on 2017-05-03 18:35
```

Thanks, I finally worked it out with a support at Microsoft. The solution is to send a POST to the API with sendInvitationMessage = false. This gives you a RedeemURL. You can then build your own email to send to your customer with this redeemURL for them to register through your AAD.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-05-05 11:30
```

Thanks for the info!

**Comment by comment-migrator** [comment]
```header
Originally posted by Aftab Shaikh on 2017-06-17 06:43
```

Hi Elio,

I am getting resource not found when i try to add guest user as a member to the Office 365 group. Any thoughts?

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-06-19 09:14
```

The invitation resource is still there, it now is even available in the v1.0 endpoint: [https://developer.microsoft...](https://developer.microsoft.com/en-us/graph/docs/api-reference/v1.0/api/invitation_post "https://developer.microsoft.com/en-us/graph/docs/api-reference/v1.0/api/invitation_post")

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-06-19 09:22
```

Just tested out the endpoint again:

![https://uploads.disquscdn.c...](https://uploads.disquscdn.com/images/51503c54244b62082c783fce2cc6bc2d3bead1840fb20b8afeef0d0979fe00ae.png)

Nothing changed, so it still works the same as before.

**Reply by comment-migrator** [reply]
```header
Originally posted by Aftab Shaikh on 2017-06-19 09:51
```

The guest user is created without any issues but adding that user as a member to the group is throwing a resource not found error. I checked Azure AD, created guest user is there then why it's complaining of not being exist when it is trying to be added as a member. Even the get user graph call for the guest user is throwing the same error.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-06-19 10:02
```

Check the ID of your new user via the users endpoint: [https://graph.microsoft.com...](https://graph.microsoft.com/v1.0/users "https://graph.microsoft.com/v1.0/users")

Could be that the ID is not the same as the one you retrieved from the invitation creation endpoint.

**Reply by comment-migrator** [reply]
```header
Originally posted by Aftab Shaikh on 2017-06-19 10:08
```

I have checked it, copy pasted the ID that I received.

The api documentation does mention about going through redemption process is must before user can use any of the resources. Is this can be the issue. Also i tried beta api, i will give a try to use v1.0.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-06-19 12:42
```

Tried it on another tenant and a new guest user. Invitation and guest user is correctly created, retrieved the new user ID via the users endpoint and added the user to an existing group. No problems at all and done via the v1.0 endpoint.

It could be some sort of hiccup on your environment. If you want, you can always create an issue at GitHub so the team can figure out what might be wrong: [https://github.com/microsof...](https://github.com/microsoftgraph/microsoft-graph-docs/issues "https://github.com/microsoftgraph/microsoft-graph-docs/issues")

**Comment by comment-migrator** [comment]
```header
Originally posted by Marc LaFleur on 2017-07-10 18:11
```

Excellent article, great work.

**Comment by comment-migrator** [comment]
```header
Originally posted by sowri mihir on 2018-03-06 17:58
```

How to add request body while doing the graph API call from C#? I am having endpoint like below, but I am not getting idea how to set the request body.  
string endpoint = string.Format("[https://graph.microsoft.com...](https://graph.microsoft.com/v1.0/groups/ "https://graph.microsoft.com/v1.0/groups/"){"+groupId+"}/members/$ref");

I am seeing request.Content option but how to set this?

**Reply by comment-migrator** [reply]
```header
Originally posted on 2018-03-14 14:39
```

Here is an article that shows you various ways of doing HTTP request (get and post): [https://sufiawan.wordpress....](https://sufiawan.wordpress.com/2015/11/13/c-several-ways-to-perform-http-get-and-post-requests/ "https://sufiawan.wordpress.com/2015/11/13/c-several-ways-to-perform-http-get-and-post-requests/")

**Comment by comment-migrator** [comment]
```header
Originally posted by Parekh Dharmesh on 2019-02-08 05:08
```

I am new to SharePoint. I am working on one project where i have to add external user to azure active directory and give him access to sharepoint online site.

I think approach you have mentioned in blog can help me. i am going to build logic app to do this and logic app will be triggered by posting http request from my web application. There is one connector named plumsail, which allows me to grant permission to external user to acces sharepoint site (read only, edit, design permission).

Adding user in office group means that user will be able to access sharepoint site. is it correct? can assign permission as well?

**Reply by comment-migrator** [reply]
```header
Originally posted on 2019-02-13 08:21
```

Yes by adding them as a member of a Office 365 Group they will also have access to the SharePoint site.

**Reply by comment-migrator** [reply]
```header
Originally posted by Parekh Dharmesh on 2019-03-07 11:00
```

Thanks

**Comment by comment-migrator** [comment]
```header
Originally posted by John on 2019-09-23 11:59
```

Can we add Guest user to MS teams using Graph API??

**Comment by comment-migrator** [comment]
```header
Originally posted by Douglas on 2020-03-05 15:19
```

Still a helpful post even 3 years later :)  
Question: When inviting a guest user (and creating an AzureAD account for the user), is it possible to also populate the FirstName, LastName, Department, MobilePhone properties?

**Comment by comment-migrator** [comment]
```header
Originally posted by Bahrain Moss 2007 on 2020-12-06 10:29
```

Is it the same process currently also ? Lot of changes happened now.

**Comment by comment-migrator** [comment]
```header
Originally posted by Nigel Price on 2021-04-03 18:15
```

Hi Elio - can we do this now with PowerAutomate (Flow) ? Regards Nigel

**Comment by comment-migrator** [comment]
```header
Originally posted by taylan celebioglu on 2021-06-18 17:19
```

Hi Elio, we can't get any information mail from Microsoft after user has been added to the group via Graph API. Do you know anything about this situation?

**Comment by the-knight305** [comment]
Great blog!
2 questions, 

Do you know how to retrieve the invitation status via power automate?
I need to this to wait until the guest has accepted before addind the guest to a team in Teams.

Also do you know how members in channels/teams are populated?

I checked my account and saw that I was member of a office365 group. I automated the invite process (with automate) and added the guest users to that group but they never show in our team....

thanks
